File: merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder result = new StringBuilder();
        int i = 0;
        for(; i < Math.min(word1.length(), word2.length()); i++) {
            result.append(word1.charAt(i));
            result.append(word2.charAt(i));
        }
        if (word1.length() < word2.length()) {
            for (; i < word2.length(); i++) {
                result.append(word2.charAt(i));
            }
        } else {
            for (; i < word1.length(); i++) {
                result.append(word1.charAt(i));
            }
        }
        return result.toString();
    }
}

File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        result = ""
        p1 = 0
        p2 = 0
        while p1 < len(word1) and p2 < len(word2):
            result += word1[p1]
            result += word2[p2]
            p1 += 1
            p2 += 1
        while p1 < len(word1):
            result += word1[p1]
            p1 += 1
        while p2 < len(word2):
            result += word2[p2]
            p2 += 1
        return result